\documentclass[11pt]{article}
\usepackage[utf8]{inputenc}
\usepackage{graphicx}
\usepackage{graphics}
\usepackage{lipsum}
\usepackage{caption}
\usepackage{subcaption}
\usepackage[a4paper,width=150mm,top=30mm,bottom=30mm]{geometry}
%\usepackage{minted}
\usepackage{hyperref}
\usepackage{fancyhdr}
\pagestyle{fancy}
\fancyhead{}
%\fancyhead[R]{Text}    % add text as header
\fancyfoot{}
\fancyfoot[C]{\thepage}   % add page number
\renewcommand{\headrulewidth}{0.10pt}
\renewcommand{\footrulewidth}{0.10pt}

%% \pagestyle{empty} remove header and footer for that page
%% \pagestyle{plain} add only the page number

\usepackage{biblatex}
\addbibresource{references.bib}

\begin{document}



\input{titlepage}

%\twocolumn[
%  \begin{@twocolumnfalse}
    \begin{abstract}
        \centering
        Metagenomic-assembled genomes are high-quality putative genomes that derive from experiments of genome assemby from metagenomic sequencing. They encode a great deal of knowledge and, even if it is not always possible to link them to a known bacterial species, they can be useful to help characterize other unknown single-genome bins. For this project, a set of putative genomes was analyzed and characterized as \textit{Adlercreutzia equolifaciens}. The core genome for this set was then determined and aligned and some phylogenetic structure were built but no evident grouping was identified with a clear link to the host data. 
    \end{abstract}
%  \end{@twocolumnfalse}
%]

% \parencite{}

%\vspace*{1cm}


\input{sections/introduction}
\input{sections/methods}
\input{sections/results/01descriptionTaxonomicAssignment.tex}
\input{sections/results/02genomeAnnotation.tex}
\input{sections/results/03pangenomeAnalysis.tex}
\input{sections/results/04phylogeneticStructureHostData.tex}
\input{sections/conclusion}


\clearpage
\appendix
\section*{Supplementary data}
\input{sections/appendix}



\clearpage
\printbibliography

\end{document}
