* Introduction:
	* Motivation: obtain information about a uSGB, starting from a set of ~30 HQ bins (i.e.
	MAGs)
	* Which uSGB are you working with, what is your focus

* Methods:
	* What software you used for each purpose, what parameters

* Results and discussion:
	* Description of the set of bins: where do your MAGs come from, to what SGB do they
	belong, completeness and contamination
	* Genome annotation: what functions are encoded in your MAGs?
		Hypothetical/annotated proteins
	* Pangenome analysis: whatâ€™s the size of your pangenome? Is it closed or open? How
	many core and accessory genes?
	* Phylogenetic analysis and association with host data: comparison of phylogenetic
	trees based on accessory gene presence/absence or on core gene alignment. Do you
	detect clusters of strains? How do they associate with the metadata?
* Conclusion
