\clearpage
\section*{Conclusion}

In this study, 31 high quality genomes (29 MAGs and 2 reference genomes)
of the organism \emph{Adlercreutzia equolifaciens} were characterized.
The set of MAGs was derived from a pool of individuals that was heterogeneous
with respect to country of provenence, health status, diseases, and age. 

All the strains were correctly classified to the species. 
Genome annotation was performed to retrieve informations about the genomes, that 
were used to perform pangenome analysis, trough which the pangenome was found to be closed.

Subsequently, the phylogenetic structure was computed and, considering the host metadata,
no significant clusters were found between the strains. As expected, the phylogeny 
retrieved from the presence or absence of accessory genes perfomed diffently with respect
to the one obtained from the core gene alignement, with the last one to be considered
more reliable. A bigger set of genomes will allow the detection of finer clustering with respect
to host metadata.

Overall, 







