\section*{Methods}

% roary commands: roary GFF/*.gff -f roary_out â€“e -p 8 -i 95 -cd 90
% we used PRANK as engine not MAFFT

%%% What software you used for each purpose, what parameters

A total of 31 HQ genomes (29 SGBs and 2 reference genome) were used in this study.

\nocite{Tange2011a}

\subparagraph*{Taxonomic assignment}

The taxonomic characterization was performed with PhyloPhlAn v. 3.0.60 (reference). In particular, the selected tool was PhyloPhlAn metagenomic: a tool that can assign genomes and MAGs to species-level genome bins by computing their distance with a database of reference genomes. The parameters chosen for the analysis were: \textbf{-nproc 4} to set the number of cores to be used by the program, \textbf{-n 1} to set the number of best taxonomic assignments to show, \textbf{--database\_update} to ensure the tool uses the latest database version and \textbf{-d cmg2122} to set the database of markers to be used. This analysis was performed on all the sequences of the set and the tab separated file obtained as output contained the complete taxonomy, from kingdom to species, of each MAG.

\subparagraph*{Genome annotation}

The genomes were annotated with the Prokka software v. 1.13 (reference). The option \textbf{--kingdom Bacteria} was selected to specify that the genome used ad input is bacterial, since Prokka also annotates archaeal and viral genomes. This procedure was applied to all the sequences of the set. As output, a series of files were obtained for each MAG and three of them were further analyzed in this project: the annotation file (.gff) was used for the pangenome analysis, the text and tab separated files were used to visualize some statistics with R (\href{https://github.com/enricofrigoli/cmg_project/blob/main/Rscript/CMG_Rscript.md}{link to the script}).


\subparagraph*{Pangenome Analysis}

%%%% PANGENOME ANALYSIS 
The species' pangenome was obtained using Roary v.3.7.0 (citation) taking Prokka annotation files (.gff) as input. 
The following parameters were given: \textbf{-i 95} to set the minimum percentage of identity
in the blastp alignement to 95\%, \textbf{-cd 90} to set the prevalence in \% MAGs for a gene
to be considered core, \textbf{-e} to produce the core gene alignement file using PRANK, \textbf{-p 8} to
specify the number of threads. 

The number of genes and related classification were retrieved from the output file
\textit{summary\_statistics.txt} and \textit{gene\_presence\_absence.csv}; plots were generated using two
plotting script (\href{https://github.com/sanger-pathogens/Roary/blob/master/bin/create_pan_genome_plots.R}{here}
and \href{https://github.com/sanger-pathogens/Roary/blob/master/contrib/roary_plots/roary_plots.py}{here}).


\subparagraph*{Phylogenetic Structure}

The resulting core gene alignement file (.aln) obtained with Roary was processed with FastTree v.2.1.10
using \textbf{-nt} as parameter (nucleotide alignement). ITOL v.6 (\href{https://itol.embl.de/}{Interactive Tree of Life})
was used to visualize both the tree obtained from Roary (.newick) and the one obtained from the core
gene alignement file processed with FastTree (.tre).




